packages/inter-protocol/src/clientSupport.js

```diff
@@ -0,0 +1,224 @@
+    };
+  }
+
+  return harden(proposal);
+};
+
+// TODO factor out BridgeAction so that these functions just return OfferSpec
+// That they are composed into a BridgeAction is the concern of the caller.
+// Then these can be used in tests as arguments to `executeOffer()` without a bridge.
+
+/**
+ * @param {Record<string, import('@agoric/vats/tools/board-utils').BoardRemote>} brands
+ * @param {{ offerId: string, wantMinted: number, giveCollateral: number, collateralBrandKey: string }} opts
+ * @returns {import('@agoric/smart-wallet/src/offers.js').OfferSpec}
+ */
+export const makeOpenOffer = (brands, opts) => {
+  const proposal = makeVaultProposal(brands, opts);
+
+  console.warn('vaults open give', proposal.give);
+  console.warn('vaults open want', proposal.want);
+
+  // NB: not really a Proposal because the brands are not remotes
+  // Instead they're copyRecord like  "{"boardId":"board0257","iface":"Alleged: IST brand"}" to pass through the boardId
+  // mustMatch(harden(proposal), ProposalShape);
+
+  const collateralBrand = brands[opts.collateralBrandKey];
+
+  return {
+    id: opts.offerId,
+    invitationSpec: {
+      source: 'agoricContract',
+      instancePath: ['VaultFactory'],
+      callPipe: [
+        ['getCollateralManager', [collateralBrand]],
+        ['makeVaultInvitation'],
+      ],
+    },
+    proposal,
+  };
+};
+
+/**
+ * @param {Record<string, import('@agoric/vats/tools/board-utils').BoardRemote>} brands
+ * @param {{ offerId: string, collateralBrandKey?: string, giveCollateral?: number, wantCollateral?: number, giveMinted?: number, wantMinted?: number }} opts
+ * @param {string} previousOffer
+ * @returns {import('@agoric/smart-wallet/src/offers.js').OfferSpec}
+ */
+export const makeAdjustOffer = (brands, opts, previousOffer) => {
+  // NB: not really a Proposal because the brands are not remotes
+  // Instead they're copyRecord like  "{"boardId":"board0257","iface":"Alleged: IST brand"}" to pass through the boardId
+  // mustMatch(harden(proposal), ProposalShape);
+  const proposal = makeVaultProposal(brands, opts);
+
+  return {
+    id: opts.offerId,
+    invitationSpec: {
+      source: 'continuing',
+      previousOffer,
+      invitationMakerName: 'AdjustBalances',
+    },
+    proposal,
+  };
+};
+
+/**
+ * @param {Record<string, import('@agoric/vats/tools/board-utils').BoardRemote>} brands
+ * @param {{ offerId: string, collateralBrandKey?: string, giveMinted: number }} opts
+ * @param {string} previousOffer
+ * @returns {import('@agoric/smart-wallet/src/offers.js').OfferSpec}
+ */
+export const makeCloseOffer = (brands, opts, previousOffer) => {
+  const proposal = makeVaultProposal(brands, opts);
+  console.warn('vaults close give', proposal.give);
+
+  return {
+    id: opts.offerId,
+    invitationSpec: {
+      source: 'continuing',
+      previousOffer,
+      invitationMakerName: 'CloseVault',
+    },
+    proposal,
+  };
+};
+
+export const Offers = {
```

Comment:
Should this be `VaultOffers`?

Reply:
I meant for it to be an object you can get multiple makers from. I'll add the PSM ones and use them before merging.